easybuild/easyblocks/p/python.py

```diff
@@ -127,6 +127,39 @@ def __init__(self, *args, **kwargs):
             easybuild_subdir = log_path()
             self.pythonpath = os.path.join(easybuild_subdir, 'python')
 
+    def patch_step(self, *args, **kwargs):
+        """
+        Custom patch step for Python:
+        * patch setup.py when --sysroot EasyBuild configuration setting is used
+        """
+
+        # if we're installing Python with an alternate sysroot,
+        # we need to patch setup.py which includes hardcoded paths like /usr/include and /lib64;
+        # this fixes problems like not being able to build the _ssl module ("Could not build the ssl module")
+        sysroot = build_option('sysroot')
+        if sysroot:
+            sysroot_inc_dirs, sysroot_lib_dirs = [], []
+
+            for pattern in ['include*', os.path.join('usr', 'include*')]:
+                sysroot_inc_dirs.extend(glob.glob(os.path.join(sysroot, pattern)))
+
+            if not sysroot_inc_dirs:
+                raise EasyBuildError("No include directories found in sysroot %s!", sysroot)
+
+            for pattern in ['lib*', os.path.join('usr', 'lib*')]:
+                sysroot_lib_dirs.extend(glob.glob(os.path.join(sysroot, pattern)))
+
+            if not sysroot_lib_dirs:
+                raise EasyBuildError("No lib directories found in sysroot %s!", sysroot)
+
+            regex_subs = [
+                (r"(system_include_dirs = \[).*\]", r"\1%s]" % ', '.join(["'%s'" % x for x in sysroot_inc_dirs])),
+                (r"(system_lib_dirs = \[).*\]", r"\1%s]" % ', '.join(["'%s'" % x for x in sysroot_lib_dirs])),
+            ]
+            apply_regex_substitutions('setup.py', regex_subs)
+
+        super(EB_Python, self).patch_step(*args, **kwargs)
```

Comment:
Shouldn't we do the `super()` call first in case there's a patch that fails because our customisation or we miss something contained in a patch?

Reply:
Sure, can do.